### ISSUE TITLE
[Bug] Many thinks are not working after migration from Robocop 5.8.1 (and Robotidy 4.16.0) to Robocop 6.0.0

### ISSUE THREAD

**Issue by phermann-DGL** [MarketSquare/robotframework-robocop#1358]
Hello after migrating to Robocop 6.0.0 a lot of things work any more correctly.

#### My old **.robocop** file:
```plaintext
--reports all
--exclude wrong-case-in-keyword-name
--configure too-many-calls-in-test-case:max_calls:30
--configure too-many-calls-in-keyword:max_calls:30
--configure too-long-test-case:max_len:60
--configure too-many-arguments:max_args:10
--configure line-too-long:line_length:200
--configure too-long-keyword:max_len:90
--configure file-too-long:max_lines:1000
--configure line-too-long:ignore_pattern:'*\{\"cards\":\[\],\"customers\":\[\],[\S+\s+]*'
--configure line-too-long:ignore_pattern:'*\$\.\.addresses\[\?\([\S+\s+]*'
--configure inconsistent-assignment-in-variables:assignment_sign_type:none
--configure inconsistent-assignment:assignment_sign_type:equal_sign
--configure mixed-task-test-settings:enabled:False
--configure variable-overwritten-before-usage:enabled:False
--configure no-suite-variable:enabled:False
--configure no-global-variable:enabled:False
```

with the old versions i runned it with these commands and it worked as expected.
```bash
robocop --output "output\webshop\before-tidy-robocop.log" --configure return_status:quality_gate:E=0:W=0:I=-1 "src\webshop"
robotidy "src\webshop"
robocop --output "output\webshop\after-tidy-robocop.log" --configure return_status:quality_gate:E=0:W=0:I=-1 "src\webshop"
```

#### New file robocop.toml

I have migrated my **.robocop** file to **robocop.toml**:

Note! For all the coming tests i used only this toml file.
```toml
[tool.robocop]


[tool.robocop.lint]
reports = ["version", "all", "text_file", "return_status"]
ignore = [
    "wrong-case-in-keyword-name",
    "variable-overwritten-before-usage",
    "no-suite-variable",
    "no-global-variable",
]
configure = [
    "line-too-long.line_length=200",
    "line-too-long.ignore_pattern=\"*{\\\"cards\\\":\\[\\],\\\"customers\\\":\\[\\],[\\S+\\s+]*\"",
    "line-too-long.ignore_pattern=\"*\\$\\.\\.addresses\\[\\?\\([\\S+\\s+]*\"",
    "too-many-calls-in-test-case.max_calls=30",
    "too-many-calls-in-keyword.max_calls=30",
    "too-long-test-case.max_len=60",
    "too-many-arguments.max_args=10",
    "too-long-keyword.max_len=90",
    "file-too-long.max_lines=1000",
    "inconsistent-assignment.assignment_sign_type=equal_sign",
    "inconsistent-assignment-in-variables.assignment_sign_type=none",
    "text_file.output_path=\"output/robocop.log\"",
    "return_status.quality_gate.E=0",
    "return_status.quality_gate.W=0",
    "return_status.quality_gate.I=-1",
]

[tool.robocop.format]
line_length = 200
```

### reports section has no effect
For this tests i used the command '**robocop check --verbose "./src/webshop"**'

```bash
D:\source\E2ETests>robocop check --verbose "./src/webshop"
Loaded D:\source\E2ETests\robocop.toml configuration file.
Scanning file: D:\source\E2ETests\src\webshop\resources\cancellation_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\card_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\cdc_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_api.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_azure.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_email.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_mongo_db.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\customer_test_data_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\keepass.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\rbe_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\search_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_browser.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_login.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_online_enrollment.resource
Scanning file: D:\source\E2ETests\src\webshop\stages\common_stages.resource
Scanning file: D:\source\E2ETests\src\webshop\tasks\import_secrets_from_json.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\01_login.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\02_login_mobile_app.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\03_my_beauty_cards.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\04_my_webshop.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\05_my_delivery_addresses.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\06_online_enrollment.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\07_newsletter.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\08_shopping_cart.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\__init__.robot
Scanning file: D:\source\E2ETests\src\webshop\translations\common_translation.resource

...

src\webshop\tests\07_newsletter.robot:496:16 ARG04 Undefined argument value, use *=${EMPTY} instead
     |
 494 |         ...    Get Element States
 495 |         ...    //div[@role="dialog"]//span[text()="${TRANSLATION_YOUR_CHANGE_WAS_SUCCESSFUL}"]
 496 |         ...    *=
     |                ^^ ARG04
 497 |         ...    visible
 498 |         IF    ${is_change_successfull}
     |

src\webshop\tests\07_newsletter.robot:543:16 ARG04 Undefined argument value, use *=${EMPTY} instead
     |
 541 |     Navigate To My Douglas
 542 |     Click    //a[@href="/${LANGUAGE}/account/data/edit#personal"]
 543 |     Get Url    *=    /account/data/edit#personal
     |                ^^ ARG04
 544 |     Type Text    //input[@name="streetName"]    ${TESTDATA_STREET_NAME}
 545 |     Get Text    //input[@name="streetName"]    ==    ${TESTDATA_STREET_NAME}
     |

src\webshop\tests\__init__.robot:5:21 IMP02 BuiltIn library import 'Collections' should be placed before 'OperatingSystem'
   |
 3 |
 4 | Library             OperatingSystem
 5 | Library             Collections
   |                     ^^^^^^^^^^^ IMP02
 6 | Library             KeePassLibrary
 7 | Library             JSONLibrary
   |


D:\source\E2ETests>
```

as you could see, there are no additional reports generated.

### return_status report crashes with error

For this test i use the commad '**robocop check --verbose --reports return_status  "./src/webshop"**'.

```bash
D:\source\E2ETests>robocop check --verbose --reports return_status  "./src/webshop"           
Loaded D:\source\E2ETests\robocop.toml configuration file.
Scanning file: D:\source\E2ETests\src\webshop\resources\cancellation_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\card_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\cdc_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_api.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_azure.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_email.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_mongo_db.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\customer_test_data_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\keepass.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\rbe_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\search_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_browser.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_login.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_online_enrollment.resource
Scanning file: D:\source\E2ETests\src\webshop\stages\common_stages.resource
Scanning file: D:\source\E2ETests\src\webshop\tasks\import_secrets_from_json.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\01_login.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\02_login_mobile_app.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\03_my_beauty_cards.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\04_my_webshop.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\05_my_delivery_addresses.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\06_online_enrollment.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\07_newsletter.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\08_shopping_cart.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\__init__.robot
Scanning file: D:\source\E2ETests\src\webshop\translations\common_translation.resource
src\webshop\resources\common.resource:25:16 VAR11 Variable '${TEST_NAME}' overwrites reserved variable '${TEST_NAME}'
    |
 23 |     ${has_test_name}=    Run Keyword And Return Status    Variable Should Exist    ${TEST_NAME}
 24 |     IF    not ${has_test_name}
 25 |         VAR    ${TEST_NAME}=    ${SUITE_NAME}    scope=GLOBAL
    |                ^^^^^^^^^^^^ VAR11
 26 |     END
 27 |     IF    '${KEYWORD_STATUS}' == 'FAIL'
    |

...

╭───────────────────────────────────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────────────────────────────────╮
│ C:\Users\<USER>\AppData\Local\Programs\Python\Python312\Lib\site-packages\robocop\run.py:314 in check_files                                                              │
│                                                                                                                                                                             │
│   311 │   │   overwrite_config=overwrite_config,                                                                                                                            │
│   312 │   )                                                                                                                                                                 │
│   313 │   runner = RobocopLinter(config_manager)                                                                                                                            │
│ ❱ 314 │   return runner.run()                                                                                                                                               │
│   315                                                                                                                                                                       │
│   316                                                                                                                                                                       │
│   317 @app.command(name="format")                                                                                                                                           │
│                                                                                                                                                                             │
│ ╭───────────────────────────────────────────────────────────── locals ──────────────────────────────────────────────────────────────╮                                       │
│ │            compare = None                                                                                                         │                                       │
│ │     config_manager = <robocop.config.ConfigManager object at 0x000001ED29163F20>                                                  │                                       │
│ │ configuration_file = None                                                                                                         │                                       │
│ │          configure = None                                                                                                         │                                       │
│ │       custom_rules = None                                                                                                         │                                       │
│ │    default_exclude = None                                                                                                         │                                       │
│ │    default_include = None                                                                                                         │                                       │
│ │            exclude = None                                                                                                         │                                       │
│ │          exit_zero = None                                                                                                         │                                       │
│ │       file_filters = FileFiltersOptions(include=None, default_include=None, exclude=None, default_exclude=None)                   │                                       │
│ │      force_exclude = False                                                                                                        │                                       │
│ │             gitlab = False                                                                                                        │                                       │
│ │             ignore = None                                                                                                         │                                       │
│ │ ignore_file_config = False                                                                                                        │                                       │
│ │     ignore_git_dir = False                                                                                                        │                                       │
│ │            include = None                                                                                                         │                                       │
│ │       issue_format = None                                                                                                         │                                       │
│ │           language = None                                                                                                         │                                       │
│ │      linter_config = LinterConfig(                                                                                                │                                       │
│ │                      │   configure=None,                                                                                          │                                       │
│ │                      │   select=None,                                                                                             │                                       │
│ │                      │   ignore=None,                                                                                             │                                       │
│ │                      │   issue_format=None,                                                                                       │                                       │
│ │                      │   target_version=<robocop.linter.utils.version_matching.Version object at 0x000001ED29163F80>,             │                                       │
│ │                      │   threshold=None,                                                                                          │                                       │
│ │                      │   custom_rules=None,                                                                                       │                                       │
│ │                      │   include_rules=set(),                                                                                     │                                       │
│ │                      │   exclude_rules=set(),                                                                                     │                                       │
│ │                      │   include_rules_patterns=set(),                                                                            │                                       │
│ │                      │   exclude_rules_patterns=set(),                                                                            │                                       │
│ │                      │   reports=['return_status'],                                                                               │                                       │
│ │                      │   persistent=None,                                                                                         │                                       │
│ │                      │   compare=None,                                                                                            │                                       │
│ │                      │   exit_zero=None,                                                                                          │                                       │
│ │                      │   return_result=False,                                                                                     │                                       │
│ │                      │   config_source='cli',                                                                                     │                                       │
│ │                      │   _checkers=None,                                                                                          │                                       │
│ │                      │   _project_checkers=None,                                                                                  │                                       │
│ │                      │   _rules=None                                                                                              │                                       │
│ │                      )                                                                                                            │                                       │
│ │   overwrite_config = Config(                                                                                                      │                                       │
│ │                      │   sources=['.'],                                                                                           │                                       │
│ │                      │   file_filters=FileFiltersOptions(                                                                         │                                       │
│ │                      │   │   include=None,                                                                                        │                                       │
│ │                      │   │   default_include=None,                                                                                │                                       │
│ │                      │   │   exclude=None,                                                                                        │                                       │
│ │                      │   │   default_exclude=None                                                                                 │                                       │
│ │                      │   ),                                                                                                       │                                       │
│ │                      │   linter=LinterConfig(                                                                                     │                                       │
│ │                      │   │   configure=None,                                                                                      │                                       │
│ │                      │   │   select=None,                                                                                         │                                       │
│ │                      │   │   ignore=None,                                                                                         │                                       │
│ │                      │   │   issue_format=None,                                                                                   │                                       │
│ │                      │   │   target_version=<robocop.linter.utils.version_matching.Version object at 0x000001ED29163F80>,         │                                       │
│ │                      │   │   threshold=None,                                                                                      │                                       │
│ │                      │   │   custom_rules=None,                                                                                   │                                       │
│ │                      │   │   include_rules=set(),                                                                                 │                                       │
│ │                      │   │   exclude_rules=set(),                                                                                 │                                       │
│ │                      │   │   include_rules_patterns=set(),                                                                        │                                       │
│ │                      │   │   exclude_rules_patterns=set(),                                                                        │                                       │
│ │                      │   │   reports=['return_status'],                                                                           │                                       │
│ │                      │   │   persistent=None,                                                                                     │                                       │
│ │                      │   │   compare=None,                                                                                        │                                       │
│ │                      │   │   exit_zero=None,                                                                                      │                                       │
│ │                      │   │   return_result=False,                                                                                 │                                       │
│ │                      │   │   config_source='cli',                                                                                 │                                       │
│ │                      │   │   _checkers=None,                                                                                      │                                       │
│ │                      │   │   _project_checkers=None,                                                                              │                                       │
│ │                      │   │   _rules=None                                                                                          │                                       │
│ │                      │   ),                                                                                                       │                                       │
│ │                      │   formatter=None,                                                                                          │                                       │
│ │                      │   language=None,                                                                                           │                                       │
│ │                      │   verbose=True,                                                                                            │                                       │
│ │                      │   target_version=7,                                                                                        │                                       │
│ │                      │   config_source='cli'                                                                                      │                                       │
│ │                      )                                                                                                            │                                       │
│ │         persistent = None                                                                                                         │                                       │
│ │            reports = ['return_status']                                                                                            │                                       │
│ │      return_result = False                                                                                                        │                                       │
│ │               root = None                                                                                                         │                                       │
│ │             runner = <robocop.linter.runner.RobocopLinter object at 0x000001ED29163E00>                                           │                                       │
│ │             select = None                                                                                                         │                                       │
│ │     skip_gitignore = False                                                                                                        │                                       │
│ │            sources = [WindowsPath('src/webshop')]                                                                                 │                                       │
│ │     target_version = None                                                                                                         │                                       │
│ │          threshold = None                                                                                                         │                                       │
│ │            verbose = True                                                                                                         │                                       │
│ ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                                       │
│                                                                                                                                                                             │
│ C:\Users\<USER>\AppData\Local\Programs\Python\Python312\Lib\site-packages\robocop\linter\runner.py:67 in run                                                             │
│                                                                                                                                                                             │
│    64 │   │   self.make_reports()                                                                                                                                           │
│    65 │   │   if self.config_manager.default_config.linter.return_result:                                                                                                   │
│    66 │   │   │   return self.diagnostics                                                                                                                                   │
│ ❱  67 │   │   return self.return_with_exit_code(issues_no)                                                                                                                  │
│    68 │                                                                                                                                                                     │
│    69 │   def run_check(                                                                                                                                                    │
│    70 │   │   self, model: File, file_path: Path, config: Config, in_memory_content: str | Non                                                                              │
│                                                                                                                                                                             │
│ ╭────────────────────────────────────────────────────────────────────── locals ───────────────────────────────────────────────────────────────────────╮                     │
│ │      config = Config(                                                                                                                               │                     │
│ │               │   sources=['.'],                                                                                                                    │                     │
│ │               │   file_filters=FileFiltersOptions(                                                                                                  │                     │
│ │               │   │   include=None,                                                                                                                 │                     │
│ │               │   │   default_include=frozenset({'*.resource', '*.robot'}),                                                                         │                     │
│ │               │   │   exclude=None,                                                                                                                 │                     │
│ │               │   │   default_exclude=frozenset({'venv', '.eggs', '.tox', 'dist', '.git', '.venv', '.hg', '.nox', '.direnv', '.svn'})               │                     │
│ │               │   ),                                                                                                                                │                     │
│ │               │   linter=LinterConfig(                                                                                                              │                     │
│ │               │   │   configure=[                                                                                                                   │                     │
│ │               │   │   │   'line-too-long.line_length=200',                                                                                          │                     │
│ │               │   │   │   'line-too-long.ignore_pattern="*{\\"cards\\":\\[\\],\\"customers\\":\\[\\],[\\S+\\s+]*"',                                 │                     │
│ │               │   │   │   'line-too-long.ignore_pattern="*\\$\\.\\.addresses\\[\\?\\([\\S+\\s+]*"',                                                 │                     │
│ │               │   │   │   'too-many-calls-in-test-case.max_calls=30',                                                                               │                     │
│ │               │   │   │   'too-many-calls-in-keyword.max_calls=30',                                                                                 │                     │
│ │               │   │   │   'too-long-test-case.max_len=60',                                                                                          │                     │
│ │               │   │   │   'too-many-arguments.max_args=10',                                                                                         │                     │
│ │               │   │   │   'too-long-keyword.max_len=90',                                                                                            │                     │
│ │               │   │   │   'file-too-long.max_lines=1000',                                                                                           │                     │
│ │               │   │   │   'inconsistent-assignment.assignment_sign_type=equal_sign',                                                                │                     │
│ │               │   │   │   ... +5                                                                                                                    │                     │
│ │               │   │   ],                                                                                                                            │                     │
│ │               │   │   select=[],                                                                                                                    │                     │
│ │               │   │   ignore=['wrong-case-in-keyword-name', 'variable-overwritten-before-usage', 'no-suite-variable', 'no-global-variable'],        │                     │
│ │               │   │   issue_format='{source}:{line}:{col} [{severity}] {rule_id} {desc} ({name})',                                                  │                     │
│ │               │   │   target_version=<robocop.linter.utils.version_matching.Version object at 0x000001ED29163F80>,                                  │                     │
│ │               │   │   threshold=<RuleSeverity.INFO: 'I'>,                                                                                           │                     │
│ │               │   │   custom_rules=[],                                                                                                              │                     │
│ │               │   │   include_rules=set(),                                                                                                          │                     │
│ │               │   │   exclude_rules={                                                                                                               │                     │
│ │               │   │   │   'variable-overwritten-before-usage',                                                                                      │                     │
│ │               │   │   │   'wrong-case-in-keyword-name',                                                                                             │                     │
│ │               │   │   │   'no-suite-variable',                                                                                                      │                     │
│ │               │   │   │   'no-global-variable'                                                                                                      │                     │
│ │               │   │   },                                                                                                                            │                     │
│ │               │   │   include_rules_patterns=set(),                                                                                                 │                     │
│ │               │   │   exclude_rules_patterns=set(),                                                                                                 │                     │
│ │               │   │   reports=['version', 'all', 'text_file', 'return_status', 'return_status'],                                                    │                     │
│ │               │   │   persistent=False,                                                                                                             │                     │
│ │               │   │   compare=False,                                                                                                                │                     │
│ │               │   │   exit_zero=False,                                                                                                              │                     │
│ │               │   │   return_result=False,                                                                                                          │                     │
│ │               │   │   config_source='D:\\source\\E2ETests\\robocop.toml',                                                            │                     │
│ │               │   │   _checkers=[                                                                                                                   │                     │
│ │               │   │   │   <comments.CommentChecker object at 0x000001ED291C69F0>,                                                                   │                     │
│ │               │   │   │   <comments.IgnoredDataChecker object at 0x000001ED291C6A80>,                                                               │                     │
│ │               │   │   │   <documentation.MissingDocumentationChecker object at 0x000001ED292083B0>,                                                 │                     │
│ │               │   │   │   <robocop.linter.rules.order.TestAndKeywordOrderChecker object at 0x000001ED2920AFC0>,                                     │                     │
│ │               │   │   │   <duplications.DuplicationsChecker object at 0x000001ED2920AFF0>,                                                          │                     │
│ │               │   │   │   <duplications.SectionHeadersChecker object at 0x000001ED2920A060>,                                                        │                     │
│ │               │   │   │   <errors.MissingKeywordName object at 0x000001ED2920BFB0>,                                                                 │                     │
│ │               │   │   │   <errors.ParsingErrorChecker object at 0x000001ED29268710>,                                                                │                     │
│ │               │   │   │   <errors.TwoSpacesAfterSettingsChecker object at 0x000001ED29268740>,                                                      │                     │
│ │               │   │   │   <errors.VariablesImportErrorChecker object at 0x000001ED29268800>,                                                        │                     │
│ │               │   │   │   ... +42                                                                                                                   │                     │
│ │               │   │   ],                                                                                                                            │                     │
│ │               │   │   _project_checkers=None,                                                                                                       │                     │
│ │               │   │   _rules={                                                                                                                      │                     │
│ │               │   │   │   'todo-in-comment': <comments.ToDoInCommentRule object at 0x000001ED291C6CF0>,                                             │                     │
│ │               │   │   │   'COM01': <comments.ToDoInCommentRule object at 0x000001ED291C6CF0>,                                                       │                     │
│ │               │   │   │   'missing-space-after-comment': <comments.MissingSpaceAfterCommentRule object at 0x000001ED291C6D80>,                      │                     │
│ │               │   │   │   'COM02': <comments.MissingSpaceAfterCommentRule object at 0x000001ED291C6D80>,                                            │                     │
│ │               │   │   │   'invalid-comment': <comments.InvalidCommentRule object at 0x000001ED291C6EA0>,                                            │                     │
│ │               │   │   │   'COM03': <comments.InvalidCommentRule object at 0x000001ED291C6EA0>,                                                      │                     │
│ │               │   │   │   'ignored-data': <comments.IgnoredDataRule object at 0x000001ED28F1BD70>,                                                  │                     │
│ │               │   │   │   'COM04': <comments.IgnoredDataRule object at 0x000001ED28F1BD70>,                                                         │                     │
│ │               │   │   │   'bom-encoding-in-file': <comments.BomEncodingRule object at 0x000001ED291C7050>,                                          │                     │
│ │               │   │   │   'COM05': <comments.BomEncodingRule object at 0x000001ED291C7050>,                                                         │                     │
│ │               │   │   │   ... +308                                                                                                                  │                     │
│ │               │   │   }                                                                                                                             │                     │
│ │               │   ),                                                                                                                                │                     │
│ │               │   formatter=FormatterConfig(                                                                                                        │                     │
│ │               │   │   whitespace_config=WhitespaceConfig(                                                                                           │                     │
│ │               │   │   │   space_count=4,                                                                                                            │                     │
│ │               │   │   │   indent=None,                                                                                                              │                     │
│ │               │   │   │   continuation_indent=None,                                                                                                 │                     │
│ │               │   │   │   line_ending='native',                                                                                                     │                     │
│ │               │   │   │   separator='space',                                                                                                        │                     │
│ │               │   │   │   line_length=200                                                                                                           │                     │
│ │               │   │   ),                                                                                                                            │                     │
│ │               │   │   select=[],                                                                                                                    │                     │
│ │               │   │   custom_formatters=[],                                                                                                         │                     │
│ │               │   │   configure=[],                                                                                                                 │                     │
│ │               │   │   force_order=False,                                                                                                            │                     │
│ │               │   │   allow_disabled=False,                                                                                                         │                     │
│ │               │   │   target_version=7,                                                                                                             │                     │
│ │               │   │   skip_config=SkipConfig(skip=set(), sections=set(), keyword_call=set(), keyword_call_pattern=set()),                           │                     │
│ │               │   │   overwrite=False,                                                                                                              │                     │
│ │               │   │   diff=False,                                                                                                                   │                     │
│ │               │   │   output=None,                                                                                                                  │                     │
│ │               │   │   color=False,                                                                                                                  │                     │
│ │               │   │   check=False,                                                                                                                  │                     │
│ │               │   │   reruns=0,                                                                                                                     │                     │
│ │               │   │   start_line=None,                                                                                                              │                     │
│ │               │   │   end_line=None,                                                                                                                │                     │
│ │               │   │   language=[],                                                                                                                  │                     │
│ │               │   │   languages=None,                                                                                                               │                     │
│ │               │   │   _parameters=None,                                                                                                             │                     │
│ │               │   │   _formatters=None                                                                                                              │                     │
│ │               │   ),                                                                                                                                │                     │
│ │               │   language=[],                                                                                                                      │                     │
│ │               │   verbose=True,                                                                                                                     │                     │
│ │               │   target_version=7,                                                                                                                 │                     │
│ │               │   config_source='D:\\source\\E2ETests\\robocop.toml'                                                                 │                     │
│ │               )                                                                                                                                     │                     │
│ │ diagnostics = []                                                                                                                                    │                     │
│ │       files = 28                                                                                                                                    │                     │
│ │   issues_no = 28                                                                                                                                    │                     │
│ │       model = <robot.parsing.model.blocks.File object at 0x000001ED2935BFD0>                                                                        │                     │
│ │        self = <robocop.linter.runner.RobocopLinter object at 0x000001ED29163E00>                                                                    │                     │
│ │      source = WindowsPath('D:/source/E2ETests/src/webshop/translations/common_translation.resource')                                 │                     │
│ ╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                     │
│                                                                                                                                                                             │
│ C:\Users\<USER>\AppData\Local\Programs\Python\Python312\Lib\site-packages\robocop\linter\runner.py:125 in return_with_exit_code                                          │
│                                                                                                                                                                             │
│   122 │   │   if self.config_manager.default_config.linter.exit_zero:                                                                                                       │
│   123 │   │   │   exit_code = 0                                                                                                                                             │
│   124 │   │   elif "return_status" in self.reports:                                                                                                                         │
│ ❱ 125 │   │   │   exit_code = self.reports["return_status"].exit_code                                                                                                       │
│   126 │   │   else:                                                                                                                                                         │
│   127 │   │   │   exit_code = 1 if issues_count else 0                                                                                                                      │
│   128 │   │   raise typer.Exit(code=exit_code)                                                                                                                              │
│                                                                                                                                                                             │
│ ╭───────────────────────────────────── locals ──────────────────────────────────────╮                                                                                       │
│ │ issues_count = 28                                                                 │                                                                                       │
│ │         self = <robocop.linter.runner.RobocopLinter object at 0x000001ED29163E00> │                                                                                       │
│ ╰───────────────────────────────────────────────────────────────────────────────────╯                                                                                       │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'ReturnStatusReport' object has no attribute 'exit_code'

D:\source\E2ETests>
```

### output_path in toml file not working for text_file report

For this test i used the command '**robocop check --verbose --reports text_file "./src/webshop" **'.

```bash
D:\source\E2ETests>robocop check --verbose --reports text_file "./src/webshop"  
Loaded D:\source\E2ETests\robocop.toml configuration file.
Scanning file: D:\source\E2ETests\src\webshop\resources\cancellation_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\card_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\cdc_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_api.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_azure.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_email.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_mongo_db.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\customer_test_data_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\keepass.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\rbe_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\search_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_browser.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_login.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_online_enrollment.resource
Scanning file: D:\source\E2ETests\src\webshop\stages\common_stages.resource
Scanning file: D:\source\E2ETests\src\webshop\tasks\import_secrets_from_json.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\01_login.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\02_login_mobile_app.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\03_my_beauty_cards.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\04_my_webshop.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\05_my_delivery_addresses.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\06_online_enrollment.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\07_newsletter.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\08_shopping_cart.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\__init__.robot
Scanning file: D:\source\E2ETests\src\webshop\translations\common_translation.resource
src\webshop\resources\common.resource:25:16 VAR11 Variable '${TEST_NAME}' overwrites reserved variable '${TEST_NAME}'
    |
 23 |     ${has_test_name}=    Run Keyword And Return Status    Variable Should Exist    ${TEST_NAME}
 24 |     IF    not ${has_test_name}
 25 |         VAR    ${TEST_NAME}=    ${SUITE_NAME}    scope=GLOBAL
    |                ^^^^^^^^^^^^ VAR11
 26 |     END
 27 |     IF    '${KEYWORD_STATUS}' == 'FAIL'
    |

...

src\webshop\tests\__init__.robot:5:21 IMP02 BuiltIn library import 'Collections' should be placed before 'OperatingSystem'
   |
 3 |
 4 | Library             OperatingSystem
 5 | Library             Collections
   |                     ^^^^^^^^^^^ IMP02
 6 | Library             KeePassLibrary
 7 | Library             JSONLibrary
   |



D:\source\E2ETests>cd output

D:\source\E2ETests\output>dir
 Datenträger in Laufwerk D: ist Daten
 Volumeseriennummer: D4CF-7CA0

 Verzeichnis von D:\source\E2ETests\output

27.03.2025  07:07    <DIR>          .
27.03.2025  06:04    <DIR>          ..
               0 Datei(en),              0 Bytes
               2 Verzeichnis(se), 62.465.138.688 Bytes frei

D:\source\E2ETests\output>  
```

### Complete configure section from toml file is overridden if you use --configure cli parameter

At first a command without --configure cli parameter
For this test i used the command **robocop check --verbose --reports version,text_file,all  "./src/webshop"**'.

```bash
D:\source\E2ETests>robocop check --verbose --reports version,text_file,all  "./src/webshop"
Loaded D:\source\E2ETests\robocop.toml configuration file.
Scanning file: D:\source\E2ETests\src\webshop\resources\cancellation_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\card_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\cdc_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_api.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_azure.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_email.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\common_mongo_db.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\customer_test_data_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\keepass.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\rbe_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\search_service.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_browser.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_login.resource
Scanning file: D:\source\E2ETests\src\webshop\resources\webshop_online_enrollment.resource
Scanning file: D:\source\E2ETests\src\webshop\stages\common_stages.resource
Scanning file: D:\source\E2ETests\src\webshop\tasks\import_secrets_from_json.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\01_login.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\02_login_mobile_app.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\03_my_beauty_cards.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\04_my_webshop.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\05_my_delivery_addresses.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\06_online_enrollment.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\07_newsletter.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\08_shopping_cart.robot
Scanning file: D:\source\E2ETests\src\webshop\tests\__init__.robot
Scanning file: D:\source\E2ETests\src\webshop\translations\common_translation.resource
src\webshop\resources\common.resource:25:16 VAR11 Variable '${TEST_NAME}' overwrites reserved variable '${TEST_NAME}'
    |
 23 |     ${has_test_name}=    Run Keyword And Return Status    Variable Should Exist    ${TEST_NAME}
 24 |     IF    not ${has_test_name}
 25 |         VAR    ${TEST_NAME}=    ${SUITE_NAME}    scope=GLOBAL
    |                ^^^^^^^^^^^^ VAR11
 26 |     END
 27 |     IF    '${KEYWORD_STATUS}' == 'FAIL'
    |

...

src\webshop\resources\webshop_browser.resource:32:12 DEPR06 Create Dictionary used instead of VAR
    |
 30 |     Register Keyword To Run On Failure    On Failure
 31 |     ${HAR_FILE_PATH}=    Set Variable    ${OUTPUT_DIR}${/}networklog-${SUITE NAME}.json
 32 |     ${har_config}=
    |            ^^^^^^^^^^^^^^^^^ DEPR06
 33 |     ...    Create Dictionary
 34 |     ...    path=${HAR_FILE_PATH}
    |

Report generated by Robocop version: 6.0.0

Processed 28 files from which 10 files contained issues.

Issues by ID:
ARG04 [E] (undefined-argument-value)      : 13
COM01 [W] (todo-in-comment)               : 6
VAR02 [I] (unused-variable)               : 5
IMP02 [W] (builtin-imports-not-sorted)    : 2
VAR11 [W] (overwriting-reserved-variable) : 1
DEPR06 [I] (replace-create-with-var)      : 1

Found 28 issues: 9 WARNINGs, 6 INFOs, 13 ERRORs.

Reported: 2025-03-27 07:27:38 +0100

Scan finished in 1.255s.

D:\source\E2ETests>
```

Now with one --configure parameter to set the text_file output path.
For this test i used the command **robocop check --verbose --reports version,text_file,all -c text_file.output_path=output/robocop.log  "./src/webshop"**'.

```bash
...

src\webshop\tests\__init__.robot:85:120 LEN08 Line is too long (172/120)
    |
 83 |     ${s_msg}=    Replace String    ${SUITE MESSAGE}    "    \\"
 84 |     ${message}=    Catenate
 85 |     ...    {"testId":"${TEST_ID}","testRunName":${{ json.dumps("""${TESTRUNNAME}""") }},"date":"${date}","country":"${COUNTRY}","language":"${LANGUAGE}","stage":"${STAGE}",
    |                                                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ LEN08
 86 |     ...    "testName":"${EMPTY}","testStatus":"${EMPTY}","testMessage":"${EMPTY}","testDescription":"${EMPTY}",
 87 |     ...    "failedSteps":{},"failedTestsWithSteps":${{ json.dumps(&{FAILED_TESTS_WITH_STEPS}) }},
    |

src\webshop\tests\__init__.robot:93:1 LEN03 Keyword 'Load Azure Servicebus Credentials' has too many keywords inside (14/10)
    |
 93 | Load Azure Servicebus Credentials
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ LEN03
 94 |     [Documentation]    Loads the service bus credentials from keepass file.
 95 |     TRY
    |


Report generated by Robocop version: 6.0.0

Processed 28 files from which 26 files contained issues.

Issues by ID:
LEN08 [W] (line-too-long)                 : 49
LEN03 [W] (too-many-calls-in-keyword)     : 33
LEN01 [W] (too-long-keyword)              : 13
ARG04 [E] (undefined-argument-value)      : 13
LEN04 [W] (too-long-test-case)            : 10
LEN07 [W] (too-many-arguments)            : 9
LEN06 [W] (too-many-calls-in-test-case)   : 8
COM01 [W] (todo-in-comment)               : 6
VAR02 [I] (unused-variable)               : 5
IMP02 [W] (builtin-imports-not-sorted)    : 2
LEN28 [W] (file-too-long)                 : 2
VAR11 [W] (overwriting-reserved-variable) : 1
DEPR06 [I] (replace-create-with-var)      : 1

Found 152 issues: 133 WARNINGs, 6 INFOs, 13 ERRORs.

Reported: 2025-03-27 09:08:20 +0100

Scan finished in 0.810s.

D:\source\E2ETests>cd output

D:\source\E2ETests\output>dir
 Datenträger in Laufwerk D: ist Daten
 Volumeseriennummer: D4CF-7CA0

 Verzeichnis von D:\source\E2ETests\output

27.03.2025  09:08    <DIR>          .
27.03.2025  06:04    <DIR>          ..
27.03.2025  09:08            20.699 robocop.log
               1 Datei(en),         20.699 Bytes
               2 Verzeichnis(se), 62.465.032.192 Bytes frei

D:\source\E2ETests\output>
```
The report is generated, but as you could see the hole configure block from toml file is ignored.

For this test i have remove the '**"text_file.output_path=\"output/robocop.log\"",**' Line from configure block in toml file.
For this test i used the command **robocop check --verbose --reports version,text_file,all -c text_file.output_path=output/robocop.log  "./src/webshop"**'.

```bash
D:\source\E2ETests>robocop check --verbose --reports version,text_file,all -c text_file.output_path=output/robocop.log  "./src/webshop"

...

src\webshop\tests\__init__.robot:85:120 LEN08 Line is too long (172/120)
    |
 83 |     ${s_msg}=    Replace String    ${SUITE MESSAGE}    "    \\"
 84 |     ${message}=    Catenate
 85 |     ...    {"testId":"${TEST_ID}","testRunName":${{ json.dumps("""${TESTRUNNAME}""") }},"date":"${date}","country":"${COUNTRY}","language":"${LANGUAGE}","stage":"${STAGE}",
    |                                                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ LEN08
 86 |     ...    "testName":"${EMPTY}","testStatus":"${EMPTY}","testMessage":"${EMPTY}","testDescription":"${EMPTY}",
 87 |     ...    "failedSteps":{},"failedTestsWithSteps":${{ json.dumps(&{FAILED_TESTS_WITH_STEPS}) }},
    |

src\webshop\tests\__init__.robot:93:1 LEN03 Keyword 'Load Azure Servicebus Credentials' has too many keywords inside (14/10)
    |
 93 | Load Azure Servicebus Credentials
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ LEN03
 94 |     [Documentation]    Loads the service bus credentials from keepass file.
 95 |     TRY
    |


Report generated by Robocop version: 6.0.0

Processed 28 files from which 26 files contained issues.

Issues by ID:
LEN08 [W] (line-too-long)                 : 49
LEN03 [W] (too-many-calls-in-keyword)     : 33
LEN01 [W] (too-long-keyword)              : 13
ARG04 [E] (undefined-argument-value)      : 13
LEN04 [W] (too-long-test-case)            : 10
LEN07 [W] (too-many-arguments)            : 9
LEN06 [W] (too-many-calls-in-test-case)   : 8
COM01 [W] (todo-in-comment)               : 6
VAR02 [I] (unused-variable)               : 5
IMP02 [W] (builtin-imports-not-sorted)    : 2
LEN28 [W] (file-too-long)                 : 2
VAR11 [W] (overwriting-reserved-variable) : 1
DEPR06 [I] (replace-create-with-var)      : 1

Found 152 issues: 133 WARNINGs, 6 INFOs, 13 ERRORs.

Reported: 2025-03-27 09:04:02 +0100

Scan finished in 1.459s.

D:\source\E2ETests>cd output

D:\source\E2ETests\output>dir
 Datenträger in Laufwerk D: ist Daten
 Volumeseriennummer: D4CF-7CA0

 Verzeichnis von D:\source\E2ETests\output

27.03.2025  09:02    <DIR>          .
27.03.2025  06:04    <DIR>          ..
27.03.2025  09:04            20.699 robocop.log
               1 Datei(en),         20.699 Bytes
               2 Verzeichnis(se), 62.465.056.768 Bytes frei

D:\source\E2ETests\output>
```
The report is generated, but as you could see the hole configure block from toml file is ignored.

**Comment by bhirsz** [comment]
I see following bugs:

1. return_status not usable - fixed in https://github.com/MarketSquare/robotframework-robocop/pull/1359
2. configure completely overwriting configuraton instead of joining (created #1360)
3. reports = ["version", "all", "text_file", "return_status"] somehow was not token into account when generating report (I will use this issue as base for this bug)
4. I also see that some lines are colorized unnecessarily, I will check it later (like <DIR> etc)

**Comment by bhirsz** [comment]
I will release 6.0.1 today which addresses findings from this thread
